internal/provider/util_test.go

```diff
@@ -0,0 +1,84 @@
+package provider
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	tfjson "github.com/hashicorp/terraform-json"
+)
+
+func Test_resourceSchema(t *testing.T) {
+	cases := []struct {
+		name                 string
+		schemas              map[string]*tfjson.Schema
+		providerShortName    string
+		templateFileName     string
+		expectedSchema       *tfjson.Schema
+		expectedResourceName string
+	}{
+		{
+			"provider short name matches schema name",
+			map[string]*tfjson.Schema{
+				"http": {},
+			},
+			"http",
+			"http.md.tmpl",
+			&tfjson.Schema{},
+			"http",
+		},
+		{
+			"provider short name does not match schema name",
+			map[string]*tfjson.Schema{
+				"http": {},
+			},
+			"tls",
+			"http.md.tmpl",
+			nil,
+			"",
+		},
+		{
+			"provider short name concatenated with template file name matches schema name",
+			map[string]*tfjson.Schema{
+				"tls_cert_request": {},
+			},
+			"tls",
+			"cert_request.md.tmpl",
+			&tfjson.Schema{},
+			"tls_cert_request",
+		},
+		{
+			"provider short name concatenated with template file name does not match schema name",
+			map[string]*tfjson.Schema{
+				"tls_cert_request": {},
+			},
+			"tls",
+			"not_found.md.tmpl",
+			nil,
+			"",
+		},
+		{
+			"provider short name concatenated with same template file name matches schema name",
+			map[string]*tfjson.Schema{
+				"tls_tls": {},
+			},
+			"tls",
+			"tls.md.tmpl",
+			&tfjson.Schema{},
+			"tls_tls",
+		},
```

Comment:
Nit: While Go allows you to omit field names, I personally find it a little more difficult to read the structs once the struct definition is "beyond the fold" since its based on ordering. e.g. my brain is like what is the 3rd field defined as "tls" here? I have to scroll up, see providerShortName, scroll back down

Reply:
Fair point. I forget that GoLand is doing this for me:

![image](https://user-images.githubusercontent.com/4332332/173022825-8a0a3928-7aee-4806-ad18-a9c2e7649c21.png)

Have added field names and refactored to use `map[string]struct{}` for the test cases.